File: 35. Search Insert Position/Solution.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # return bisect.bisect_left(nums, target)
        if target <= nums[0]: return 0
        if target > nums[-1]: return len(nums)
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target: return mid
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        return left

File: 35. Search Insert Position/Solution.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        /**
        int res = Arrays.binarySearch(nums, target);
        if (res >= 0) return res;
        return -1 * (res + 1);
         */
        int n = nums.length, left = 0, right = n - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target) return mid;
            if (nums[mid] < target) {
                left++;
            } else {
                right--;
            }
        }
        return left;
    }
}